src/core/defcomp.jl

```diff
@@ -24,7 +24,7 @@ function _generate_run_func(comp_name, module_name, args, body)
     func = :(
         global function $(func_name)($(p)::Mimi.ComponentInstanceParameters, 
                                      $(v)::Mimi.ComponentInstanceVariables, 
-                                     $(d)::Mimi.DimValueDict, 
+                                     $(d)::Mimi.DimValueNamedTuple, 
```

Comment:
I would just not type these arguments at all here, you can just leave this as `$(d)`. The type of this is not required for dispatch, so really it has no purpose here. That is true for all of the functions below.